### PR TITLE
add filter to data_project in byName func and update the tests

### LINKED ISSUE #80: `data "env0_project"` should not take archived projects into account
1. Manually create two projects in env0 called "TestProject"
2. Archive one of those projects
3. Create `env0_project` `data` in a TF file with `name = "TestProject"`:
```
data "env0_project" "test_project" {
  name = "TestProject"
}
```
4. `terraform plan` fails with "Found multiple Projects for name"

This `data` should completely ignore archived projects, as they are technically "destroyed"

### PR DESCRIPTION
**Pull request by samuel-br**
### Issue & Steps to Reproduce / Feature Request
[//]: <> (choose “resolves” “fixes” or “closes” and put link for the issue)
fixes #80 
### Solution

add filter to by_name func in data_project that filter project by isArchived

### CODE REVIEW

File: env0/data_project_test.go [hunk 1]
```diff
@@ -17,6 +17,15 @@ func TestProjectDataSource(t *testing.T) {
...
 	}
 
+	archivedProject := client.Project{
+		Id:          "id0",
+		Name:        "my-project-1",
+		CreatedBy:   "env0",
+		Role:        "role0",
+		Description: "A project's description",
+		IsArchived:  true,
+	}
```
Comment (eranelbaz) [inline env0/data_project_test.go:27]: please use the values from `project` where only `Id` and `IsArchived` should be different

**Review by eranelbaz — COMMENTED** [review]
looking good
lets just fix the test to a DRYer one

File: env0/data_project_test.go (hunk 1 above)
Comment (samuel-br) [reply env0/data_project_test.go]: done.

Comment (eranelbaz) [reply env0/data_project_test.go]: please comment with the commit hash

Comment (samuel-br) [reply env0/data_project_test.go]: 4c9a58c88842bf9b52c5b8683cd5a713b54d602f

File: env0/data_project_test.go [hunk 2]
```diff
@@ -17,6 +17,15 @@ func TestProjectDataSource(t *testing.T) {
 		Description: "A project's description",
 	}
 
+	archivedProject := client.Project{
+		Id:          "otherId",
+		Name:        project.Id,
```
Comment (eranelbaz) [inline env0/data_project_test.go:22]: wrong attribute
```suggestion
		Name:        project.Name,
```

File: env0/data_project_test.go (hunk 1 above)
Comment (samuel-br) [reply env0/data_project_test.go]: 58e077d89f144f056ed038ff261bb52cf10ce0e5

**Review by eranelbaz — APPROVED** [review]

File: env0/data_project_test.go (hunk 2 above)
Comment (samuel-br) [reply env0/data_project_test.go]: 58e077d89f144f056ed038ff261bb52cf10ce0e5

**Pull request merged**